### PR TITLE
adjust docker-lint to use hadolint

### PR DESCRIPTION
**Pull request by szaimen**
Close https://github.com/nextcloud/all-in-one/discussions/3225

todo: test clamav

### CODE REVIEW

**Comment by szaimen** [comment]
Thinks to fix:
```
./Containers/mastercontainer/Dockerfile:19 DL3019 info: Use the `--no-cache` switch to avoid the need to use `--update` and remove `/var/cache/apk/*` when done installing packages
./Containers/mastercontainer/Dockerfile:19 DL3047 info: Avoid use of wget without progress bar. Use `wget --progress=dot:giga <url>`. Or consider using `-q` or `-nv` (shorthands for `--quiet` or `--no-verbose`).
./Containers/mastercontainer/Dockerfile:[19](https://github.com/nextcloud/all-in-one/actions/runs/5953115782/job/16146625375?pr=3231#step:4:20) SC2086 info: Double quote to prevent globbing and word splitting.
./Containers/mastercontainer/Dockerfile:19 DL4006 warning: Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check
./Containers/nextcloud/Dockerfile:19 DL3019 info: Use the `--no-cache` switch to avoid the need to use `--update` and remove `/var/cache/apk/*` when done installing packages
./Containers/nextcloud/Dockerfile:19 DL4006 warning: Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check
./Containers/nextcloud/Dockerfile:19 SC2086 info: Double quote to prevent globbing and word splitting.
./Containers/redis/Dockerfile:6 DL4006 warning: Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check
./Containers/postgresql/Dockerfile:8 DL4006 warning: Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check
./Containers/notify-push/Dockerfile:6 DL4006 warning: Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check
./Containers/talk/Dockerfile:47 DL4006 warning: Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check
./Containers/talk-recording/Dockerfile:11 DL3042 warning: Avoid use of cache directory with pip. Use `pip install --no-cache-dir <package>`
./Containers/talk-recording/Dockerfile:11 DL4006 warning: Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check
./Containers/apache/Dockerfile:15 DL4006 warning: Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check
./Containers/clamav/Dockerfile:6 DL4006 warning: Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check
./Containers/clamav/Dockerfile:6 SC2002 style: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.
./Containers/mastercontainer/Dockerfile:19 DL3019 info: Use the `--no-cache` switch to avoid the need to use `--update` and remove `/var/cache/apk/*` when done installing packages
./Containers/mastercontainer/Dockerfile:19 DL3047 info: Avoid use of wget without progress bar. Use `wget --progress=dot:giga <url>`. Or consider using `-q` or `-nv` (shorthands for `--quiet` or `--no-verbose`).
./Containers/mastercontainer/Dockerfile:19 SC2086 info: Double quote to prevent globbing and word splitting.
./Containers/mastercontainer/Dockerfile:19 DL4006 warning: Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check
./Containers/nextcloud/Dockerfile:19 DL3019 info: Use the `--no-cache` switch to avoid the need to use `--update` and remove `/var/cache/apk/*` when done installing packages
./Containers/nextcloud/Dockerfile:19 DL4006 warning: Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check
./Containers/nextcloud/Dockerfile:19 SC[20](https://github.com/nextcloud/all-in-one/actions/runs/5953115782/job/16146625375?pr=3231#step:4:21)86 info: Double quote to prevent globbing and word splitting.
./Containers/redis/Dockerfile:6 DL4006 warning: Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check
./Containers/postgresql/Dockerfile:8 DL4006 warning: Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check
./Containers/notify-push/Dockerfile:6 DL4006 warning: Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check
./Containers/talk/Dockerfile:47 DL4006 warning: Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check
./Containers/talk-recording/Dockerfile:11 DL3042 warning: Avoid use of cache directory with pip. Use `pip install --no-cache-dir <package>`
./Containers/talk-recording/Dockerfile:11 DL4006 warning: Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check
./Containers/apache/Dockerfile:15 DL4006 warning: Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check
./Containers/clamav/Dockerfile:6 DL4006 warning: Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check
./Containers/clamav/Dockerfile:6 SC2002 style: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.
```

File: .github/workflows/docker-lint.yml [hunk 1]
```diff
@@ -27,28 +27,24 @@ jobs:
...
-          wget https://github.com/replicatedhq/dockerfilelint/pull/201.patch -O /usr/local/lib/node_modules/dockerfilelint/201.patch
-          CURRENT_DIR=$PWD
-          cd /usr/local/lib/node_modules/dockerfilelint/
-          git apply 201.patch
-          cd $CURRENT_DIR
-          cat << RULES > ./.dockerfilelintrc
-          rules:
-            sudo_usage: off
-          RULES
+          sudo wget https://github.com/hadolint/hadolint/releases/download/v2.12.0/hadolint-Linux-x86_64 -O /usr/bin/hadolint
```
Comment (szaimen) [inline .github/workflows/docker-lint.yml:32]: should always get the latest version automatically like in update-helm.sh

Comment (Zoey2936) [reply .github/workflows/docker-lint.yml:32]: https://github.com/hadolint/hadolint/releases/latest/download/hadolint-Linux-x86_64

Comment (Zoey2936) [reply .github/workflows/docker-lint.yml:32]: did you seen that?

File: .github/workflows/docker-lint.yml [hunk 2]
```diff
@@ -27,28 +27,24 @@ jobs:
...
+            # DL3018 warning: Pin versions in apk add. Instead of `apk add <package>` use `apk add <package>=<version>`
+            # DL3008 warning: Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`
+            # DL3002 warning: Last USER should not be root
+            # DL3003 warning: Use WORKDIR to switch to a directory
+            # DL3004 error: Do not use sudo as it leads to unpredictable behavior. Use a tool like gosu to enforce root
+            hadolint "$file" --ignore DL3018 --ignore DL3008 --ignore DL3002 --ignore DL3003 --ignore DL3004 | tee -a ./hadolint.log
           done
-          if grep "^Issues: [0-9]" ./dockerfilelint.log; then
+          cat ./hadolint.log
+          if grep -q "DL[0-9]\+" ./hadolint.log; then
```
Comment (Zoey2936) [inline .github/workflows/docker-lint.yml:48]:
```suggestion
          if grep -q ""DL[0-9]\+\|SC[0-9]\+"" ./hadolint.log; then
```

File: .github/workflows/docker-lint.yml (hunk 1 above)
Comment (szaimen) [reply .github/workflows/docker-lint.yml:32]: Yed, thanks! :)

**Comment by szaimen** [comment]
Todo: 
- rebase after https://github.com/nextcloud/all-in-one/pull/3243

**Comment by Zoey2936** [comment]
what part of this PR is with the "help wanted" label meant?

**Comment by szaimen** [comment]
> what part of this PR is with the "help wanted" label meant?

It means here that I would like to get some help fixing the remaining issues :)

File: .github/workflows/docker-lint.yml (hunk 1 above)
Comment (szaimen) [reply .github/workflows/docker-lint.yml:32]:
```suggestion
          sudo wget https://github.com/hadolint/hadolint/releases/latest/download/hadolint-Linux-x86_64 -O /usr/bin/hadolint
```

**Pull request merged**